### PR TITLE
Homework 17 polymorphism encapsulation

### PR DESCRIPTION
**Pull request by nkstlrv**

### CODE REVIEW

File: HW_17_Polymorphism/task_1_hw17.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+class Cat(Animal):
+    def talk(self):
+        return f"{__class__.__name__} says meow"
+
+
+c_1 = Dog().talk()
+c_2 = Cat().talk()
+
+
+def talking_func(animal):
```
Comment (SergeyTovstonog) [inline HW_17_Polymorphism/task_1_hw17.py:20]: as per task, function should accept instance class, not str.

File: HW_17_Polymorphism/task_2_hw17.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    def new_book(self, book_name, year, author_name, country, birthday):
+        new_author = Author(author_name, country, birthday, [book_name])
+        new_book = Book(book_name, year, new_author)
+        for book in self.l_books:
+            if book.b_name == book_name:
+                print("This book is already in the library")
+                break
+        for author in self.l_authors:
+            if book.b_name != book_name and author.a_name == author_name:
```
Comment (SergeyTovstonog) [inline HW_17_Polymorphism/task_2_hw17.py:71]: same for Authors

File: HW_17_Polymorphism/task_3_hw17.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+class Fraction:
+
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+
+    def __add__(self, other):
```
Comment (SergeyTovstonog) [inline HW_17_Polymorphism/task_3_hw17.py:7]: these methods need to return `Fraction`
return `Fruction(a, b)`

Comment (nkstlrv) [reply HW_17_Polymorphism/task_3_hw17.py:10]: Made some fixes

File: HW_17_Polymorphism/task_1_hw17.py (hunk 1 above)
Comment (nkstlrv) [reply HW_17_Polymorphism/task_1_hw17.py:16]: Made some fixes

File: HW_17_Polymorphism/task_2_hw17.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        return f"An instance of the {__class__.__name__} class with name {self.l_name}"
+
+    def __repr__(self):
+        return f"{self.l_name}"
+
+    def new_book(self, book_name, year, author_name, country, birthday):
+        new_author = Author(author_name, country, birthday, [book_name])
+        new_book = Book(book_name, year, new_author)
+        for book in self.l_books:
+            if book.b_name == book_name:
```
[the remark this answers is not in the document]
Comment (nkstlrv) [reply HW_17_Polymorphism/task_2_hw17.py:67]: Will work on it tomorrow

File: HW_17_Polymorphism/task_2_hw17.py (hunk 2 above)
Comment (nkstlrv) [reply HW_17_Polymorphism/task_2_hw17.py:71]: Will work on it tomorrow

**Comment by nkstlrv** [comment]
Task 2 fixes are also made

**Review by SergeyTovstonog — APPROVED** [review]
Great work!

**Pull request merged**